app/controllers/course/assessment/submission_question/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+class Course::Assessment::SubmissionQuestion::CommentsController < Course::Assessment::Controller
+  helper Course::Assessment::Submission::SubmissionsHelper.name.sub(/Helper$/, '')
+
+  # PostsConcern MUST be included after loading subimssion_question
+  load_and_authorize_resource :submission_question,
+                              class: Course::Assessment::SubmissionQuestion.name
+  include Course::Discussion::PostsConcern
+
+  delegate :discussion_topic, to: :@submission_question
+
+  def create
+    @submission_question.class.transaction do
+      @post.title = @assessment.title
+      # Set parent as the topologically last pre-existing post, if it exists.
+      # @post is in @submission_question.posts, so we filter out @post, which has no id yet.
+      if @submission_question.posts.length > 1
+        @post.parent = @submission_question.posts.ordered_topologically.flatten.select(&:id).last
```

Comment:
Can I request to move this into a separate method, there is quite a few things going on in this method. So it will look something like this: 
`@post.parent = last_post_from(@submission_question) if @submission_question.posts.length > 1`